### PR TITLE
Tdl 18579 add unit test cases

### PR DESCRIPTION
**Pull request by kethan1122**
# Description of change

- add unittest cases for http error handling, context fns, metadata fns etc..
- fixed expected record count in test_bookmarks.py for integration tests (been failing from long time)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by savan-chovatiya — CHANGES_REQUESTED** [review]
Add pytest coverage thing in config.yml so code coverage can be visible in artifacts.
Reference: https://github.com/singer-io/tap-sailthru/blob/main/.circleci/config.yml#L29-L38

File: tests/unittests/test_auth_discovery.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    def test_basic_auth_no_access_401(self):
+        '''
+            Verify exception is raised for no access(401) error code for basic auth
+            do the assertions inside exception block
+        '''
+        args = Args()
+        try:
+            tap_zendesk_chat.discover(args.config)
+        except HTTPError as e:
```
[the remark this answers is not in the document]
Comment (kethan1122) [reply tests/unittests/test_auth_discovery.py]: changed

**Comment by kethan1122** [comment]
> Add pytest coverage thing in config.yml so code coverage can be visible in artifacts. Reference: https://github.com/singer-io/tap-sailthru/blob/main/.circleci/config.yml#L29-L38

File: tests/unittests/test_auth_discovery.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (savan-chovatiya) [reply tests/unittests/test_auth_discovery.py]: Please update all the try-except in unit tests.

**Review by savan-chovatiya — APPROVED** [review]

**Pull request merged**